backend/src/app/tasks/file_gc.clj

```diff
@@ -162,7 +164,73 @@
   (let [sql (str "delete from file_thumbnail "
                  " where file_id=? and revn < ?")
         res (db/exec-one! conn [sql file-id revn])]
-    (l/debug :hint "delete file thumbnails" :file-id file-id :total (:next.jdbc/update-count res))))
+    (when-not (zero? (:next.jdbc/update-count res))
+      (l/debug :hint "delete file thumbnails" :file-id file-id :total (:next.jdbc/update-count res)))))
+
+(def ^:private
+  sql:retrieve-client-files
+  "select f.data, f.modified_at
+     from file as f
+     left join file_library_rel as fl on (fl.file_id = f.id)
+    where fl.library_file_id = ?
+      and f.modified_at < ?
+      and f.deleted_at is null
+    order by f.modified_at desc
+    limit 1")
+
+(defn- retrieve-client-files
+  "search al files that use the given library.
+   Returns a sequence of file-data (only reads database rows one by one)."
+  [conn library-id]
+  (let [get-chunk (fn [cursor]
+                    (let [rows (db/exec! conn [sql:retrieve-client-files library-id cursor])]
+                      [(some-> rows peek :modified-at)
+                       (map #(->> % :data blob/decode) rows)]))]
+
+    (d/iteration get-chunk
+                 :vf second
+                 :kf first
+                 :initk (dt/now))))
+
+(defn- clean-deleted-components!
+  "Performs the garbage collection of unreferenced deleted components."
+  [conn library-id library-data]
+  (let [find-used-components-file
+        (fn [components file-data]
+          ; Find what of the components are used in the file.
+          (d/filterm #(ctf/used-in? file-data library-id (second %) :component)
```

Comment:
I think you don't need to use `d/filterm` here (see other related comments)

Reply:
ok